tiledb/sm/fragment/single_fragment_info.h

```diff
@@ -267,9 +272,38 @@ class SingleFragmentInfo {
     return ss.str();
   }
 
-  /** Returns the array schema name. */
-  const std::string& array_schema_name() const {
-    return array_schema_name_;
+  /** Accessor to the fragment size. */
+  uint64_t& fragment_size() {
+    return fragment_size_;
+  }
+
+  /** Accessor to the metadata pointer. */
+  shared_ptr<FragmentMetadata>& meta() {
+    return meta_;
+  }
+
+  /** Accessor to the metadata pointer. */
+  NDRange& non_empty_domain() {
+    return non_empty_domain_;
+  }
+
+  void set_info_from_meta() {
+    if (meta_ == nullptr) {
+      throw std::logic_error("Cannot set info from empty fragment metadata.");
+    }
+    uri_ = meta_->fragment_uri();
+    version_ = meta_->format_version();
+    sparse_ = !meta_->dense();
+    timestamp_range_ = meta_->timestamp_range();
+    cell_num_ = meta_->cell_num();
+    has_consolidated_footer_ = meta_->has_consolidated_footer();
+    array_schema_name_ = meta_->array_schema_name();
+    non_empty_domain_ = meta_->non_empty_domain();
+    expanded_non_empty_domain_ = non_empty_domain_;
+    if (!sparse_) {
```

Comment:
This type of initialization is not intuitive, it isn't a direct consequence of the deserialiation process, but rather a hack to make up for SingleFragmentInfo not being c41 compliant, I've encountered a lot of such cases too in the global writer serialization work. Can you please add a comment that is trying to explain why this initialization is needed?

Reply:
So, in this case I don't think that it's a c41 compliance issue as SingleFragmentInfo inializes expanded_non_empty_domain_ in its constructor. It was actually my bad for using the default constructor and then breaking c41 for this class by introducing this method to do 2-phase construction. I refactored `single_fragment_info_from_capnp` to use the full constructor instead and got rid of this method. Good catch!